crates/build/src/docker.rs

```diff
@@ -0,0 +1,607 @@
+        .context("Invalid start time")?
+        .as_secs() as i64;
+
+    maybe_println!(
+        verbosity,
+        " {} {}",
+        format!("{build_steps}").bold(),
+        "Started the build inside the container"
+            .bright_green()
+            .bold(),
+    );
+
+    let options = Some(WaitContainerOptions {
+        condition: "not-running",
+    });
+
+    let mut wait_stream = client.wait_container(&container_id, options);
+    let handle_error = |r: Result<ContainerWaitResponse, Error>| -> Result<()> {
+        let response = match r {
+            Ok(v) => v,
+            Err(e) => {
+                // sometimes we cannot remove the file due to privileged access
+                let _ = std::fs::remove_file(build_result_path);
+                anyhow::bail!("{}. Execution failed!", e.to_string())
+            }
+        };
+        if response.status_code != 0 {
+            anyhow::bail!("Execution failed! Status code: {}.", response.status_code);
+        }
+        Ok(())
+    };
+    if verbosity.is_verbose() {
+        let spinner_style =
+            ProgressStyle::with_template(" {spinner:.cyan.bold} {wide_msg}")?
+                .tick_chars("⠁⠂⠄⡀⢀⠠⠐⠈ ");
+
+        let pb = ProgressBar::new(1000);
+        pb.enable_steady_tick(Duration::from_millis(100));
+        pb.set_style(spinner_style);
+        pb.set_message("Build is being executed...");
+        while let Some(r) = wait_stream.next().await {
+            let res = handle_error(r);
+            if let Some(e) = res.err() {
+                let err_logs: Vec<LogOutput> = client
+                    .logs::<String>(
+                        &container_id,
+                        Some(LogsOptions {
+                            follow: false,
+                            stdout: true,
+                            stderr: true,
+                            since: start_time,
+                            ..Default::default()
+                        }),
+                    )
+                    .filter_map(|l| l.ok())
+                    .collect()
+                    .await;
+                // cargo dumps compilation status together with other logs
+                // we need to filter our those messages
+                let rex = regex::Regex::new(r"\[=*> \]")?;
+                let err_string = err_logs
+                    .iter()
+                    .filter_map(|l| {
+                        if let LogOutput::Console { message } = l {
+                            let msg = String::from_utf8(message.to_vec()).unwrap();
+                            let msg =
+                                msg.split('\r').filter(|m| !rex.is_match(m)).collect();
+                            Some(msg)
+                        } else {
+                            None
+                        }
+                    })
+                    .collect::<Vec<String>>()
+                    .join("");
+                anyhow::bail!("{}\n{}", e, err_string);
```

Comment:
This doesn't appear to work correctly, but at least we can see the error now which is better than before:

![image](https://github.com/paritytech/cargo-contract/assets/75586/1f520452-45fb-4fb3-be10-7961defee9f0)

Reply:
Ah, I need manually filter out logs, because everything is dumped into `CONSOLE` output

Reply:
Should be fixed by the recent commit